Изменения

FletchDS opened the pull request.

FletchDS committed:
Изменения

Antony8720 commented on src/main/java/ru/hogwarts/school/model/Student.java:
```diff
@@ -16,6 +14,11 @@ public class Student {
     private String name;
     private int age;
 
+    @ManyToOne
```
Установлена феч-стратегия FetchType.LAZY.

Antony8720 commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -24,8 +23,20 @@ public Faculty getFaculty(@PathVariable Long id){
     @GetMapping
-    public Collection<Faculty> getFacultiesOfSpecificColor(@RequestParam String color){
-        return facultyService.getFacultiesOfSpecificColor(color);
+    public ResponseEntity getFacultiesOfSpecificColor(@RequestParam(required = false) String color,
```
Если поглубже изучить аннотацию @RestController, то она оборачивает (с помощью аннотации @ResponseBody) тело каждого метода в ResponseEntity. Таким образом нам нет смысла явно использовать этот класс в сигнатуре метода, и избавившись от него, наш код станет более удобочитаемым. Внеси, пожалуйста, правки в код.

Antony8720 commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -18,27 +16,37 @@ public StudentController(StudentService studentService) {
     @GetMapping
-    public Collection<Student> getStudentsOfSpecificAge(@RequestParam("age") int age){
-        return studentService.getAllStudentsOfSpecificAge(age);
+    public ResponseEntity getStudentsOfSpecificAge(@RequestParam("age") Integer age,
```
Если поглубже изучить аннотацию @RestController, то она оборачивает (с помощью аннотации @ResponseBody) тело каждого метода в ResponseEntity. Таким образом нам нет смысла явно использовать этот класс в сигнатуре метода, и избавившись от него, наш код станет более удобочитаемым. Внеси, пожалуйста, правки в код.

FletchDS committed:
Изменения

FletchDS committed:
Изменения
Добавлена работа с потоками

FletchDS committed:
Изменения
-Убран лишний код
-getFacultiesOfSpecificColor переименован в getFaculties
-getStudentsOfSpecificAge переименован в getStudents
-В конструкторы Faculty.java и Student.java добавлены новые аргументы классов
-В метод getStudents контроллера StudentController.java добавлена возможность получения всех студентов, если не указывать возраст
